### PR TITLE
Fix issue #17914 : Add python bindings to neuron derivation and minor…

### PR DESCRIPTION
**Pull request by adevress**
###### Motivation for this change

 Solve  issue #17914
- add python bindings support
- normalize prefix / path following nix conventions
- minor fixes 
###### Things done
- [x] Tested using sandboxing
  ([nix.useChroot](http://nixos.org/nixos/manual/options.html#opt-nix.useChroot) on NixOS,
    or option `build-use-chroot` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
  - [ ] NixOS
  - [ ] OS X
  - [x] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

… fixes
- Add support for python bindings
- make neuron respect standard pythonpath prefix
- force exec_prefix == prefix to respect standard nix file hierarchy
- normalize indentation
- propagate dependencies necessary for nrniv_makefile usage

### CODE REVIEW

File: pkgs/applications/science/biology/neuron/default.nix [hunk 1]
```diff
@@ -1,33 +1,68 @@
...
+                      ]
+                    ++ (if mpi != null then ["--with-mpi" "--with-paranrn"] 
+                        else ["--without-mpi"]);
+                        
+                        
+  postInstall = ''
+    ## standardise python neuron install dir if any
+    if [[ -d $out/lib/python ]]; then
+        LOCAL_PYTHONVER="$(${python}/bin/python -c "from distutils.sysconfig import get_python_version; print(get_python_version())")"
+        LOCAL_PYTHONDIR="$out/lib/python''${LOCAL_PYTHONVER}"
```
Comment (FRidh) [inline pkgs/applications/science/biology/neuron/default.nix]: `python` has some passthru attributes that you could use here

**Comment by adevress** [comment]
@FRidh Fixed. I switched to python passthru attributes

**Comment by FRidh** [comment]
nix/store/gd28ivv40gs4mjjmkf6zlzb71ydvx3y5-binutils-2.26.1/bin/ld: cannot find -libverbs

**Comment by adevress** [comment]
@FRidh  I have seen. It is a transient problem with the openmpi package.

Neuron do not use nor depend on libibverbs. 

I will re-triger the build to see if the problem has been fixed or not

**Comment by FRidh** [comment]
no need, I'll build it locally

**Comment by adevress** [comment]
@FRidh I just rebased with fork. If the problem has been fixed into openmpi, it should disappear.

**Comment by FRidh** [comment]
before you updated the PR I reproduced the error. Testing now with the update.

**Comment by adevress** [comment]
The problem still exist on trunk.

It comes from the fact openmpi ship the libtool .la files when it should not

consequently, libtool in neuron try to add the indirect dependency (libibverbs) when it links with openmpi when this one is not part of its propagated dependency.

libibverbs should not be part of the propagated dependency.
Get rids of the *.la files shipped by openmpi should solve the issue.

 Libtool *.la files are harmful to package managers as a general rule

**Comment by FRidh** [comment]
@adevress please open an issue about that. It would be nice if we could fix that for the next release.

**Comment by adevress** [comment]
@FRidh  Yep. I am giving a try with a manual strip of the *.la files.
if it works, I will open a issue and a pull request for openmpi.

**Comment by FRidh** [comment]
Pushed to master, thanks!

**Pull request closed without merging**